SCons/Node/FS.py

```diff
@@ -3216,9 +3207,11 @@ def get_csig(self):
 
         csig = self.get_max_drift_csig()
         if csig is None:
-
             try:
-                if self.get_size() < File.md5_chunksize:
+                size = self.get_size()
+                if size == -1:
+                    contents = SCons.Util.NOFILE
+                elif size < File.md5_chunksize:
                     contents = self.get_contents()
```

Comment:
Does this need to be get_text_contents()? get_contents -> bytes..

Reply:
Hmmm.... it's been that way all along (that is, you picked on the one unchanged line in this sequence :) ).

Reply:
the signature function operates on bytes; `MD5Signature` calls `to_bytes` first to make sure, so it would work either way.